code-challenges.js

```diff
@@ -9,60 +9,201 @@
+// .length is another built in method that counts the number of indexes in an array
+// combine both arrays using .concat, then add .length at the end to count the indexes.
+// place this in a console log to display the output in the terminal
+
+const padres1984WorldSeriesRuns = [2, 5, 2, 2, 4];
+const padres1998WorldSeriesRuns = [6, 3, 5, 3];
+
+console.log(padres1984WorldSeriesRuns.concat(padres1998WorldSeriesRuns).length);
 
-const padres1984WorldSeriesRuns = [2, 5, 2, 2, 4]
-const padres1998WorldSeriesRuns = [6, 3, 5, 3]
 // Expected output: 9
+// actual output: 9
 
 // --------------------3) Create the code that will reverse the letters of a string using the test variable provided below.
 
 // Pseudo code:
+// .reverse() is a built in method that will reverse an ARRAY, but we have a string
+// use .split("") after currentCohort to convert the string into an array, followed by .reverse() to to reverse the indexes of said array, then use .join("") to convert the array back into a string. the output will be a reverse of the original string
+// call on the above in a console log to display the output in terminal
+
+const currentCohort = "Alpha 2023";
+
+console.log(currentCohort.split("").reverse().join(""));
 
-const currentCohort = "Alpha 2023"
 // Expected output: "3202 ahplA"
+// actual output : "3202 ahplA"
 
 // --------------------4) Create the code that will return the last index of the given value from the array using the test variables provided below.
 
 // Pseudo code:
+// two methods to this problem.
+// i can use the .lastIndexOf() built in method at the end of of variable containing an array to return the last index of a given value
+// console.log(myNumbers.lastIndexOf(givenValue1)); output: 7
+// if i want to use logic to do the same thing, i need to create a function (retrieveIndex) that when called upon initiates a for loop that looks at each index of the array and compares it to a given value, this for loop should start at the end of the array and work towards the beginning (8-0) ending at the 0 index and incrementing 1 index at a time. within the for loop of my retrieveIndex function, i also want to have a conditional statement that if a given index matches the paramater provided, i want to return that index.
+// create a console.log() with each variable to display the output in the terminal.
 
-const myNumbers = [13, 34, 42, 5, 5, 10, 27, 42, 10]
-const givenValue1 = 42
-// Expected output: 7
+const myNumbers = [13, 34, 42, 5, 5, 10, 27, 42, 10];
 
-const givenValue2 = 10
+const givenValue1 = 42;
+// Expected output: 7
+const givenValue2 = 10;
 // Expected output: 8
 
+// console.log(myNumbers.lastIndexOf(givenValue1));
+// console.log(myNumbers.lastIndexOf(givenValue2));
+
+const retrieveIndex = (value) => {
+  for (let i = myNumbers.length - 1; i >= 0; i--) {
+    if (myNumbers[i] === value) {
+      return i;
+    }
+  }
+};
+console.log(retrieveIndex(givenValue1));
```

Comment:
Wow! This is a really cool approach! If you wanted to continue to refactor this, you could explore what would happen if the value isn't in the array.